web_interface/blueprints/api_v3.py

```
@@ -1341,6 +1341,57 @@ def get_system_version():
+        ).stdout.strip()
+        remote_sha = subprocess.run(
+            ['git', 'rev-parse', 'origin/main'],
+            capture_output=True, text=True, timeout=5, cwd=project_dir
+        ).stdout.strip()
+
+        if local_sha == remote_sha:
+            data = {'status': 'success', 'update_available': False,
+                    'local_sha': local_sha[:8], 'remote_sha': remote_sha[:8]}
+        else:
+            log_result = subprocess.run(
+                ['git', 'log', 'HEAD..origin/main', '--oneline'],
```

Ruff (0.15.10)

[error] Starting a process with a partial executable path

(S607)